handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java

```diff
@@ -34,96 +39,116 @@
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assume.assumeNoException;
 
+@RunWith(Parameterized.class)
 public class JdkSslEngineTest extends SSLEngineTest {
-    private static final String PREFERRED_APPLICATION_LEVEL_PROTOCOL = "my-protocol-http2";
-    private static final String FALLBACK_APPLICATION_LEVEL_PROTOCOL = "my-protocol-http1_1";
-    private static final String APPLICATION_LEVEL_PROTOCOL_NOT_COMPATIBLE = "my-protocol-FOO";
+    public enum ProviderType {
+        NPN_DEFULT {
```

Comment:
@johnou I'm not sure I understand your comment ... is there something that should change here?